influence_benchmark/stats/preferences_per_iteration.py

```diff
@@ -8,43 +8,48 @@
 
 def load_trajectories(trajectory_path: Path) -> pd.DataFrame:
     # Read all trajectories from files
-    traj_timesteps = pd.concat([pd.read_json(file, lines=True) for file in trajectory_path.glob("[0-9]*.jsonl")])
+    traj_timestep_df = pd.concat([pd.read_json(file, lines=True) for file in trajectory_path.glob("[0-9]*.jsonl")])
 
     # Calculate expected preference
-    traj_timesteps["timestep_reward"] = traj_timesteps["preferences"].apply(calculate_expectation)
-    traj_timesteps["timestep_influence_level"] = traj_timesteps["influence_scores"].apply(calculate_expectation)
-    return traj_timesteps
+    traj_timestep_df["timestep_reward"] = traj_timestep_df["preferences"].apply(calculate_expectation)
+    traj_timestep_df["timestep_influence_level"] = traj_timestep_df["influence_scores"].apply(calculate_expectation)
+    return traj_timestep_df
+
+
+def compute_average_traj_rewards(traj_timestep_df):
+    # Average over turns, will include num_envs * num_initial_states * num_trajs_per_initial_state rows
+    avg_rewards_df = (
+        traj_timestep_df.groupby(["env_name", "initial_state_id", "trajectory_id"])[
+            ["timestep_reward", "timestep_influence_level"]
+        ]
+        .mean()
+        .reset_index()
+        .rename(columns={"timestep_reward": "traj_mean_rew", "timestep_influence_level": "traj_mean_infl"})
+    )
+    return avg_rewards_df
 
 
 def get_best_worst_n_trajectories(traj_path: Path, num_chosen_trajs: int) -> Tuple[List[Dict], List[Dict]]:
-    top_n = get_func_n_trajectories(traj_path, num_chosen_trajs, pd.DataFrame.nlargest)
-    bottom_n = get_func_n_trajectories(traj_path, num_chosen_trajs, pd.DataFrame.nsmallest)
-    return top_n, bottom_n
+    top_n_df = get_func_n_trajectories(traj_path, num_chosen_trajs, pd.DataFrame.nlargest)
+    bottom_n_df = get_func_n_trajectories(traj_path, num_chosen_trajs, pd.DataFrame.nsmallest)
+    return top_n_df, bottom_n_df
```

Comment:
might be misleading to call it df when it is not a dataframe?

Reply:
It is, in fact, a df. I think the previous name was misleading, and I was very confused about this initially